v2/tools/asoctl/internal/importing/arm_resource_importer.go

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package importing
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/klog/v2"
+
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/internal/reflecthelpers"
+	"github.com/Azure/azure-service-operator/v2/internal/version"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/pkg/names"
+
+	armruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/runtime"
+	azruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+)
+
+// ARMResourceImporter is an importer for ARM based resources.
+// Create one using the factory method on ResourceImporter.
+type ARMResourceImporter struct {
+	ResourceImporter
+	client               *azruntime.Pipeline
+	serviceConfiguration cloud.ServiceConfiguration
+}
+
+// Import imports the specified ARM resource, returning the imported resource.
+func (ri *ARMResourceImporter) Import(ctx context.Context, armID string) (*ResourceImportResult, error) {
+	// Parse armID into a more useful form
+	id, err := arm.ParseResourceID(armID)
+	if err != nil {
+		return nil, err // arm.ParseResourceID already returns a good error, no need to wrap
+	}
+
+	// Create a blank object into which we capture the current state of the resource
+	obj, err := ri.createBlankObjectFromID(id)
+	if err != nil {
+		// Error doesn't need additional context
+		return nil, err
+	}
+
+	importable, ok := obj.(genruntime.ImportableARMResource)
+	if !ok {
+		return nil, errors.Errorf(
+			"unable to create blank resource, expected %s to identify an importable ARM object", armID)
+	}
+
+	status, err := ri.getStatus(ctx, armID, importable)
+	if err != nil {
+		return nil, err
+	}
+
+	err = importable.InitializeSpec(status)
+	if err != nil {
+		return nil, errors.Wrapf(err, "setting status on Kubernetes resource for resource %s", armID)
+	}
+
+	return &ResourceImportResult{
+		resources: []genruntime.MetaObject{importable},
+	}, nil
+}
+
+// getStatus gets the status of the resource with the specified ID from ARM
+func (ri *ARMResourceImporter) getStatus(ctx context.Context, armID string, armMeta genruntime.ARMMetaObject) (genruntime.ConvertibleStatus, error) {
```

Comment:
Same comment as below -- we have a lot of these REST semantics in `generic_client` already. You should be able to access it at `v2/internal/genericarmclient` and use it here, just like we're already using `"github.com/Azure/azure-service-operator/v2/internal/version"` in root.go of `asoctl`.

Reply:
Done.